lib/ably/realtime/channel/channel_manager.rb

```diff
@@ -111,16 +111,12 @@ def fail_messages_awaiting_ack(error, options = {})
         end
       end
 
-      # When a channel becomes detached, suspended or failed,
+      # When a channel becomes suspended or failed,
       # all queued messages should be failed immediately as we don't queue in
       # any of those states
       def fail_queued_messages(error)
         error = Ably::Exceptions::MessageDeliveryFailed.new("Queued messages on channel '#{channel.name}' in state '#{channel.state}' will never be delivered") unless error
         fail_messages_in_queue connection.__outgoing_message_queue__, error
```

Comment:
> `fail_messages_in_queue connection.__outgoing_message_queue__, error`

Not something you added in this pr, but: am I misunderstanding something, or does this fail every message in the connection queue if any channel becomes suspended or failed? That's not right, surely

Reply:
No, that is not correct, see https://github.com/ably/ably-ruby/pull/166/files/f7a524e00e64a47cf5c9d974dbcf621f5b82307a#diff-b8c5e1edac0c00c8db3f3e6ebcc8e749R125, only the messages for that channel.   However, based on what we agree on https://github.com/ably/docs/pull/470, I'll most likely need to modify this...